### PR TITLE
Download Svg Button

### PR DESCRIPTION
**Pull request by ChaituVR**
<sub>
By opening a pull request, I certify that I hold the intellectual property of the code I am submitting,
and I am granting the initial authors of WBO a perpetual, worldwide, non-exclusive, royalty-free, and irrevocable license to this code.
</sub>

Added a download button to export as svg
image
below grid icon:
image

### CODE REVIEW

**Comment by droundy** [comment]
I tried using this button (which would be awesome), but found that it caused my math (from the math tool pull request) to disappear.  it also gave some weird errors when I opened the svg in firefox or google-chrome, which may or may not be related.

**Comment by AliKarpuzoglu** [comment]
Hi, 

could it be that the math features are added in another canvas?
as far as I could tell the drawing was just added to one element with the id 'canvas'

File: client-data/tools/download/download.js [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+        var styleNode = document.createElement("style");
+        styleNode.innerHTML = "rect { fill:none; } path, line {fill: none;stroke-linecap: round; stroke-linejoin: round;}";
+        canvas.appendChild(styleNode);
+        var element = document.createElement('a');
+        element.setAttribute('href', 'data:image/svg+xml;charset=utf-8,' + encodeURIComponent(canvas.outerHTML));
+        element.setAttribute('download', "file.svg");
+        element.style.display = 'none';
+        document.body.appendChild(element);
+        element.click();
+        document.body.removeChild(element);
```
Comment (vinuvvk88) [inline client-data/tools/download/download.js:40]: In this function, you have added styleNode to the canvas in line 32 & 33. After downloading the file, that styleNode needs to be removed from the canvas. 

+ canvas.removeChild(styleNode);

If not we have so many other rendering issues after downloading the file. For example, after downloading, look at the opacity & grid icons - these won't work.

File: client-data/tools/download/download.js [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    }
+
+    Tools.add({ //The new tool
+        "name": "Download",
+        "shortcut": "d",
+        "listeners": {},
+        "icon": "tools/download/download.svg",
```
Comment (vinuvvk88) [inline client-data/tools/download/download.js:47]: The download icon is too bold in comparison to other icons. To keep it uniform the boldness needs to be reduced.

File: client-data/tools/download/download.js [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+ *
+ * @licend
+ */
+
+(function download() { //Code isolation
+
+    var canvas = document.getElementById('canvas');
+    function downloadFile(evt) {
+        var styleNode = document.createElement("style");
+        styleNode.innerHTML = "rect { fill:none; } path, line {fill: none;stroke-linecap: round; stroke-linejoin: round;}";
```
Comment (vinuvvk88) [inline client-data/tools/download/download.js:32]: -        styleNode.innerHTML = "rect { fill:none; } path, line {fill: none;stroke-linecap: round; stroke-linejoin: round;}";
+        styleNode.innerHTML = "rect, ellipse { fill:none; } path, line {fill: none;stroke-linecap: round; stroke-linejoin: round;}";

Add "fill:none" for ellipse as well.

**Review by vinuvvk88 — COMMENTED** [review]
Found a couple of bugs and here's my review to fix those.

**Comment by vinuvvk88** [comment]
> I tried using this button (which would be awesome), but found that it caused my math (from the math tool pull request) to disappear. it also gave some weird errors when I opened the svg in firefox or google-chrome, which may or may not be related.

This is becasue of the issues I noted in my review: https://github.com/lovasoa/whitebophir/pull/122#pullrequestreview-545414211

**Comment by DanielHabenicht** [comment]
Any updates on this? I would love this feature and implement the missing comments.

**Pull request merged**